src/pages/member/MProfile.js

```diff
@@ -0,0 +1,597 @@
+    }, 
+    infoBox: {
+        paddingLeft: 70,
+        paddingBottom: 10,
+        display: 'flex', 
+        flexdirection: 'row', 
+    }, 
+    infoLabel: {
+        fontStyle: 'normal', 
+        fill: 'solid', 
+        fontSize: 24,
+        fontWeight: 'bold', 
+        color: '#AEC4F4',
+    }, 
+    infoValue: {
+        fontStyle: 'normal', 
+        fill: 'solid', 
+        fontSize: 20,
+        color: '#AEC4F4', 
+    },
+    pencilIcon: {
+        color: '#AEC4F4'
+    }, 
+    icon: {
+        marginLeft: 'auto',
+        paddingRight: 70, 
+    },
+    submitButton: {
+        color: '#AEC4F4', 
+        variant: "contained"
+    }, 
+    button: {
+        marginLeft: 'auto',
+        paddingRight: 70, 
+    }, 
+
+    eventCard: {
+        height: 495,
+        marginLeft: 40, 
+    }, 
+
+    membershipCard: {
+        marginLeft: 40,
+    },
+    
+    membershipCardContent: {
+        display: "flex", 
+        flexDirection: "row", 
+        justifyContent: "space-around",
+        paddingTop: 30,
+    }, 
+    membershipCardEventNumber: {
+        fontWeight: "bold",
+        fontSize: 20,  
+    }, 
+    membershipCardYear: {
+        fontWeight: "bold",
+        fontSize: 20, 
+    }, 
+    membershipCardEventNumberText: {
+        display: "flex", 
+        flexDirection: "row",
+        paddingTop: 30,
+
+    }, 
+    eventsContent: {
+        display: "flex", 
+        flexDirection: "column",
+        justifyContent: "space-between",
+        paddingLeft: 25,
+        paddingTop: 30
+    }, 
+    eventLabel: {
+        color: '#96FF50', 
+        fontSize: 26, 
+        fontWeight: "bold", 
+        paddingTop: 25,
+    }, 
+    eventValue: {
+        color: '#AEC4F4',
+        fontSize: 26,
+    }
+}));
+
+function MemberProfile(props) {
+    const [isEditingID, setIsEditingID] = React.useState(false); 
+    const [isEditingFaculty, setIsEditingFaculty] = React.useState(false);
+    const [isEditingEmail, setIsEditingEmail] = React.useState(false);
+    const [isEditingDiet, setIsEditingDiet] = React.useState(false);
+    const [ID, setID] = React.useState(props.user.id); 
+    const [Faculty, setFaculty] = React.useState(props.user.faculty); 
+    const [Email, setEmail] = React.useState(props.user.email);
+    const [Diet, setDiet] = React.useState(props.user.diet); 
+    const [eventsAttended, setEventsAttended] = React.useState();
+    const [recentEvent, setRecentEvent] = React.useState({});
+    const [favouriteEventIDs, setFavouriteEventIDs] = React.useState([]); 
+    const [favouriteEventName1, setFavouriteEventName1] = React.useState({})
+    const [favouriteEventName2, setFavouriteEventName2] = React.useState({}) 
+    const Fname = React.useState(props.user.fname); 
+    const Lname = React.useState(props.user.lname); 
+    const Year = React.useState(props.user.level); 
+    const Gender = React.useState(props.user.gender); 
+
+    const body = {
+        email: Email,
+        fname: Fname,
```

Comment:
i dont thikn you need a Fname constant declared at the top, you could just do fname: props.user.fname here

Reply:
I changed body fields to props.user.____ where I could, but for some reason I can't pass Year as props.user.level